lib/archethic/contracts.ex

```diff
@@ -143,4 +143,25 @@ defmodule Archethic.Contracts do
   """
   @spec stop_contract(binary()) :: :ok
   defdelegate stop_contract(address), to: Loader
+
+  @doc """
+  Determines if the contract's execution is valid for the given transaction
+  """
+  @spec valid_contract_execution?(Contract.t(), Transaction.t(), list(Transaction.t())) ::
+          boolean()
+  def valid_contract_execution?(contract = %Contract{}, tx = %Transaction{}, calls \\ []) do
+    case Interpreter.execute(:transaction, contract, tx, [tx | calls]) do
+      {:ok, _} ->
+        true
+
+      _ ->
+        false
```

Comment:
Do we really want to transform a "invalid condition constraint" into "invalid contract execution"?

Reply:
Why the invalid condition constraint will be raise ?

Reply:
I think it's ok